AppWidget/README.md

```diff
@@ -1,396 +1,81 @@
-* [`maxResizeWidth`](https://developer.android.com/reference/android/appwidget/AppWidgetProviderInfo#maxResizeWidth) and [`maxResizeHeight`](https://developer.android.com/reference/android/appwidget/AppWidgetProviderInfo#maxResizeHeight): Defines the maximum size the launcher allows the user to resize the widget.
+# Getting Started
 
-The following XML describes how to use the sizing attributes.
+This sample uses the Gradle build system. To build this project, use the
+"gradlew build" command or use "Import Project" in Android Studio.
 
-[`xml/app_widget_info_checkbox_list.xml`](app/src/main/res/xml/app_widget_info_checkbox_list.xml\#L38)
+The sample contains 3 type of widgets:
 
-```xml
-<appwidget-provider
-  android:maxResizeWidth="240dp"
-  android:maxResizeHeight="220dp"
-  android:minWidth="180dp"
-  android:minHeight="180dp"
-  android:minResizeWidth="180dp"
-  android:minResizeHeight="180dp"
-  android:targetCellWidth="3"
-  android:targetCellHeight="3"
-  ... />
-```
+- ToDo list: showcasing how to create a list of items
+- Buttons: showcasing how to use the new CompoundedButtons and handle click events
+- Weather: showcasing a "real-life" weather widget.
 
-<img src="screenshots/widget_resizing.png"
-      alt="Screenshot of a widget being resized"
-      title="Screenshot of a widget being resized" />
+Those are implemented in two packages:
 
-### Provide responsive layouts [(docs)](https://developer.android.com/about/versions/12/features/widgets#provide-responsive-layouts)
+- [rv](app/src/main/java/com/example/android/appwidget/rv) for RemoteViews implementation
+- [glance](app/src/main/java/com/example/android/appwidget/glance) for Glance implementation
 
-If the layout needs to change depending on the size of the widget, we recommend creating a small set of layouts, each valid for a range of sizes. (If this isn’t possible, another option is to provide layouts based on the exact widget size at runtime.)
+In addition, the [MainActivity](app/src/main/java/com/example/android/appwidget/MainActivity.kt)
+showcases how to request the launcher to "pin" an appwidget.
 
-Implementing this feature allows for smoother scaling and overall better system health; this is because the system doesn't have to wake up the app every time it displays the widget in a different size.
+<img src="screenshots/todo-gif.gif" width="256px"
+alt="TODO widget showcase"
+title="Showcase of the TODO widget implementation" />
+<img src="screenshots/weather-gif.gif" width="256px"
+alt="Weather widget showcase"
+title="Showcase of the Weather widget implementation" />
+<img src="screenshots/images-gif.gif" width="256px"
+alt="Images widget showcase"
+title="Showcase of the Images widget implementation" />
 
-The following code examples show how to provide a list of layouts.
+> More showcasing resources in the [screenshots folder](screenshots)
 
-[`ListAppWidget.kt`](app/src/main/java/com/example/android/appwidget/ListAppWidget.kt\#L94)
+## Run Configuration
```

Comment:
Replace the text in lines 52 through 59 with the following:

## Run a new configuration

When creating a new run configuration, it's important to ensure that the widget is recreated without launching a new activity. Follow these steps to do so:

1. In Android Studio, select **Run > Edit Configurations**.
2. In the **Run/Debug Configurations** dialog, select **Always install with package manager...**.
3. Under **Launch Options**, in the **Launch** menu, select **Nothing**.